src/main/java/org/prebid/server/spring/config/SettingsConfiguration.java

```diff
@@ -217,6 +226,103 @@ public DatabasePeriodicRefreshService ampDatabasePeriodicRefreshService(
         }
     }
 
+    @Configuration
+    @ConditionalOnProperty(prefix = "settings.s3", name = {"accounts-dir", "stored-imps-dir", "stored-requests-dir"})
+    static class S3SettingsConfiguration {
+
+        @Component
+        @ConfigurationProperties(prefix = "settings.s3")
+        @ConditionalOnProperty(prefix = "settings.s3", name = {"accessKeyId", "secretAccessKey"})
+        @Validated
+        @Data
+        @NoArgsConstructor
+        protected static class S3ConfigurationProperties {
+            @NotBlank
+            private String accessKeyId;
+            @NotBlank
+            private String secretAccessKey;
+            /**
+             * If not provided AWS_GLOBAL will be used as a region
+             */
+            private String region;
+            @NotBlank
+            private String endpoint;
+            @NotBlank
+            private String bucket;
+            @NotBlank
+            private String accountsDir;
+            @NotBlank
+            private String storedImpsDir;
+            @NotBlank
+            private String storedRequestsDir;
+            @NotBlank
+            private String storedResponsesDir;
+        }
+
+        @Bean
+        S3AsyncClient s3AsyncClient(S3ConfigurationProperties s3ConfigurationProperties) throws URISyntaxException {
+            final AwsBasicCredentials credentials = AwsBasicCredentials.create(
+                    s3ConfigurationProperties.getAccessKeyId(),
+                    s3ConfigurationProperties.getSecretAccessKey());
+            final String awsRegionName = s3ConfigurationProperties.getRegion();
+            final Region awsRegion = awsRegionName != null
+                    ? Region.of(s3ConfigurationProperties.getRegion())
+                    : Region.AWS_GLOBAL;
+            return S3AsyncClient
+                    .builder()
+                    .credentialsProvider(StaticCredentialsProvider.create(credentials))
+                    .endpointOverride(new URI(s3ConfigurationProperties.getEndpoint()))
+                    .region(awsRegion)
+                    .build();
```

Comment:
@muuki88 could we also include a `forcePathStyle(true)` parameter, possibly as a `settings.s3.force-path-style` setting? Starting with version 2.18, the [AWS SDK for Java v2 includes new virtual bucket addressing](https://github.com/aws/aws-sdk-java-v2/blob/2be163a018b15515125a0ca20cf011de26e5fa73/CHANGELOG.md#L10:~:text=Notably%2C%20there%20are%20some%20changes%20in%20S3%3A%20when%20including%20and%20endpoint%20override%2C%20the%20SDK%20will%20use%20virtual%20bucket%20addressing%20as%20long%20as%20the%20bucket%20is%20a%20valid%20DNS%20label.%20This%20is%20different%20from%20the%20previous%20behavior%20where%20this%20only%20happens%20if%20the%20endpoint%20override%20begins%20with%20%27s3%27.).

Old:
`http://tests.com:8080/foo`

New:
`http://foo.tests.com:8080`

This parameter switches it back to the old behavior. It would be nice to have such flexibility in the code, and it would also simplify changes in functional tests.

Reply:
Good catch! I added this option.